lms/djangoapps/teams/models.py

```diff
@@ -0,0 +1,98 @@
+"""
+Django models related to teams functionality.
+"""
+
+from django.contrib.auth.models import User
+from django.db import models
+from django.utils.translation import ugettext_lazy
+from django.conf import settings
+from django_countries.fields import CountryField
+from .utils import slugify
+
+from xmodule_django.models import CourseKeyField
+
+
+class CourseTeam(models.Model):
+    """
+    This model represents team related info.
+    """
+
+    team_id = models.CharField(max_length=255, unique=True)
+    name = models.CharField(max_length=255)
+    is_active = models.BooleanField(default=True)
+    course_id = CourseKeyField(max_length=255, db_index=True)
+    topic_id = models.CharField(max_length=255, db_index=True, blank=True)
+    date_created = models.DateTimeField(auto_now_add=True)
+    # last_activity is computed through a query
+    description = models.CharField(max_length=1000)
+    country = CountryField(blank=True)
+    language = models.CharField(
+        max_length=16,
+        blank=True,
+        choices=settings.ALL_LANGUAGES,
+        help_text=ugettext_lazy("Optional language the team uses as ISO 639-1 code.")
+    )
+    users = models.ManyToManyField(User, db_index=True, related_name='teams', through='CourseTeamMembership')
+
+    @classmethod
+    def create(cls, name, course_id, description, topic_id=None, country=None, language=None):
+        """
+        Create a complete CourseTeam object.
+
+        Args:
+            name: Name of the team to be created
```

Comment:
Nit: the args should include types, and the descriptions should be full sentences ending with a period. See the User API for examples (or the Google docstring example site):

https://github.com/edx/edx-platform/blob/master/openedx/core/djangoapps/user_api/preferences/api.py#L27